### PR TITLE
Move detect_megacli in the megacli module

### PR DESCRIPTION
**Pull request by elfosardo**
Remove detect_megacli from detect and move it in the megacli module.
Also some refactoring of the same module to improve readability
and comply to pep8 standards.

### CODE REVIEW

File: hardware/megacli.py [hunk 1]
```diff
@@ -195,31 +198,93 @@ def ld_get_num(ctrl):
...
-            print()
-            print('Logical disk', ld_num)
-            pprint.pprint(ld_get_info(ctrl_num, ld_num))
+def detect():
+    """Detect LSI MegaRAID controller configuration."""
+    hw_lst = []
+    ctrl_num = adp_count()
+    disk_count = 0
+    global_pdisk_size = 0
+    if ctrl_num > 0:
```
Comment (ErwanAliasr1) [inline hardware/megacli.py:211]: I would invert the logic here as all the rest of the code is indented again this.
Retuning an explicit value for  ctrl_num==0 and then continue the rest of the logic with the final return

**Review by ErwanAliasr1 — CHANGES_REQUESTED** [review]

File: hardware/megacli.py (hunk 1 above)
Comment (elfosardo) [reply hardware/megacli.py:211]: yeah, good call, I'll change it

**Comment by elfosardo** [comment]
> I'm picky ... you could move hw_lst = [] at the first line and return hw_lst if ctrl_num == 0
> This way we have the same variable for both cases, if the data type change, that's easier to update.

/facepalm I'm not even sure why I didn't do that....

**Review by ErwanAliasr1 — APPROVED** [review]
Thx for the split & rework

**Pull request merged**